beacon/goclient/goclient.go

```diff
@@ -402,13 +401,21 @@ func (gc *GoClient) singleClientHooks() *eth2clienthttp.Hooks {
 // so we decided that it's best to assert that GenesisForkVersion is the same.
 // To add more assertions, we check the whole apiv1.Genesis (GenesisTime and GenesisValidatorsRoot)
 // as they should be same too.
-func (gc *GoClient) assertSameGenesisVersion(genesisVersion phase0.Version) (phase0.Version, error) {
-	if gc.genesisForkVersion != genesisVersion {
-		fmt.Printf("genesis fork version mismatch, expected %v, got %v", gc.genesisForkVersion, genesisVersion)
-		return gc.genesisForkVersion, fmt.Errorf("genesis fork version mismatch, expected %v, got %v", gc.genesisForkVersion, genesisVersion)
+func (gc *GoClient) assertSameGenesisVersion(genesis *apiv1.Genesis) (phase0.Version, error) {
+	if genesis == nil {
+		return phase0.Version{}, fmt.Errorf("genesis is nil")
 	}
 
-	return gc.genesisForkVersion, nil
+	if gc.genesis.CompareAndSwap(nil, genesis) {
+		return genesis.GenesisForkVersion, nil
+	}
+
+	expected := *gc.genesis.Load()
+	if expected.GenesisForkVersion != genesis.GenesisForkVersion {
+		return expected.GenesisForkVersion, fmt.Errorf("genesis fork version mismatch, expected %v, got %v", expected.GenesisForkVersion, genesis.GenesisForkVersion)
+	}
```

Comment:
- `gc.genesis.Load()` returns a pointer, but `expected := *gc.genesis.Load()` assumes it’s always **non-nil**.
 - If `gc.genesis.Load()` somehow returns `nil`, this would panic.

suggested fix
```go
expectedPtr := gc.genesis.Load()
if expectedPtr == nil {
    return phase0.Version{}, fmt.Errorf("unexpected nil genesis pointer")
}
expected := *expectedPtr
```

Reply:
It's not expected to be nil because of this code:

```go
	if genesis == nil {
		return phase0.Version{}, fmt.Errorf("genesis is nil")
	}

	if gc.genesis.CompareAndSwap(nil, genesis) {
		return genesis.GenesisForkVersion, nil
	}
```

If `gc.genesis` is nil, then it will be set to `genesis`. If `genesis` is nil, it will return an error

Reply:
then why do we need to Load() here?

Reply:
@moshe-blox it needs to be thread-safe because `gc.genesis` is common for all clients, so `assertSameGenesisVersion` may be called in parallel

Reply:
FYI @moshe-blox @olegshmuelov @nkryuchkov 
we have a task to use a hardcoded value instead of taking the first available client. The point is that we know which network we're using so we could just hardcode this value which is constant and simplify this flow a lot.

Reply:
@y0sher The task you mentioned has already been merged into stage: https://github.com/ssvlabs/ssv/pull/2044

Reply:
okay, I reverted this change